department-of-veterans-affairs/abd-vro
# Remove references to yoomlam's GH PAT

## What was the problem?

[Here](https://github.com/department-of-veterans-affairs/abd-vro/wiki/LHDI's-Boilerplate-Instructions#secrets) is where I was instructed (by LHDI) to use my PAT (personal access token), stored in `secrets.ACCESS_TOKEN`.

GHCR packages are associated to me personally and (I think) no one else can delete them.

Associated tickets or Slack threads:

- #767

## How does this fix it?


In most (maybe all?) cases within a GH workflow, [secrets.GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#using-the-github_token-in-a-workflow) can be used instead, so that build artifacts are not associated to yoomlam personally.

(Doesn't seem necessary so far: In other cases, use a PAT for a GitHub ["machine user" account](https://docs.github.com/en/developers/overview/managing-deploy-keys#machine-users) called `abd-vro-machine`.)

- [ ] Reminder to remove `secrets.ACCESS_TOKEN` and associated `secrets.USERNAME` from [GitHub settings](https://github.com/department-of-veterans-affairs/abd-vro/settings/secrets/actions) after this PR is merged.
   - We can't remove `secrets.ACCESS_TOKEN` from the internal repo until secrel.yml is deprecated (i.e., when secrel4prs.yml is fully functional/successful)
- [x] Reminder to replace `ACCESS_TOKEN` from [Dependabot's secrets](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Dependabot#checks-github-actions-fail-on-dependabot-generated-pr) with machine-user's PAT.

## How to test this PR
Run all modified GH workflows against this branch. See comments below for examples.
